pkg/client/client.go

```diff
@@ -233,18 +253,66 @@ func Initialize(
 		}
 	})
 
-	for _, application := range sanctionedApplications {
-		go checkStatusAndRegisterForApplication(ctx, ethereumChain, application)
+	blockCounter := hostChain.BlockCounter()
+
+	supportedApplicationsByStringID := make(map[string]chain.BondedECDSAKeepApplicationHandle)
+	tbtcApplicationHandle, err := hostChain.TBTCApplicationHandle()
+	if err != nil {
+		logger.Errorf(
+			"failed to look up on-chain tBTC application information for chain " +
+				"[%s]; tBTC application functionality will be disabled",
+		)
+	} else {
+		supportedApplicationsByStringID[tbtcApplicationHandle.ID().String()] =
+			tbtcApplicationHandle
+	}
+	for _, applicationAddress := range sanctionedApplications {
+		handle, exists := supportedApplicationsByStringID[applicationAddress.String()]
+		if exists {
+			go checkStatusAndRegisterForApplication(ctx, blockCounter, handle)
+		} else {
+			logger.Warnf(
+				"unknown sanctioned application address [%s]",
+				applicationAddress,
```

Comment:
That prints a byte mess in logs. Can we call `.String()` ? 

```suggestion
				applicationAddress.String(),
```

Reply:
Resolving as per https://github.com/keep-network/keep-ecdsa/pull/714#discussion_r586510627

Reply:
See [this conversation](https://github.com/keep-network/keep-ecdsa/pull/709#discussion_r585467417). Tldr: the next PR should be the move to chain.*ID, and should fix this without requiring explicit `.String` calls everywhere.